### PR TITLE
ENH: add kw `level` to `Styler.hide_columns`

### PR DESCRIPTION
**Pull request by attack68**
follows #42636, which added `levels` to `Styler.hide_index`. This adds same functionality.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1873,14 +1858,18 @@ def hide_index(
             self.hidden_rows = hrows  # type: ignore[assignment]
         return self
 
-    def hide_columns(self, subset: Subset | None = None) -> Styler:
+    def hide_columns(
+        self,
+        subset: Subset | None = None,
+        levels: Level | list[Level] | None = None,
```
Comment (jreback) [inline pandas/io/formats/style.py:1864]: umm didn't we change this to `level` ?

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -3139,3 +3150,37 @@ def css_calc(x, left: float, right: float, align: str):
             index=data.index,
             columns=data.columns,
         )
+
+
+def _refactor_levels(
```
Comment (jreback) [inline pandas/io/formats/style.py:3155]: can do after, but this is fairly generic and doens't belong here; we also do this elsewhere in the codebase. pls create an issue / PR if you can.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style.py:3155]: #42933

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py]: have changed this, and from the last pr (`hide_index`) also.

File: doc/source/whatsnew/v1.4.0.rst [hunk 3]
```diff
@@ -34,7 +34,7 @@ Other enhancements
 - :meth:`Series.sample`, :meth:`DataFrame.sample`, and :meth:`.GroupBy.sample` now accept a ``np.random.Generator`` as input to ``random_state``. A generator will be more performant, especially with ``replace=False`` (:issue:`38100`)
 -  Additional options added to :meth:`.Styler.bar` to control alignment and display, with keyword only arguments (:issue:`26070`, :issue:`36419`)
 - :meth:`Styler.bar` now validates the input argument ``width`` and ``height`` (:issue:`42511`)
-- Add keyword ``levels`` to :meth:`.Styler.hide_index` for optionally controlling hidden levels in a MultiIndex (:issue:`25475`)
+- Add keyword ``levels`` to :meth:`.Styler.hide_index` and :meth:`.Styler.hide_columns` for optionally controlling hidden levels in a MultiIndex (:issue:`25475`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:37]:
```suggestion
- Add keyword ``level`` to :meth:`.Styler.hide_index` and :meth:`.Styler.hide_columns` for optionally controlling hidden levels in a MultiIndex (:issue:`25475`)
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm merge on greenish

**Pull request merged**